### PR TITLE
Add notes field

### PR DESCRIPTION
**Pull request by alientian**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 1]
```diff
@@ -37,7 +39,7 @@ public class AddCommandParser implements Parser<AddCommand> {
     public AddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_COMPANY_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ROLE,
-                        PREFIX_DEADLINE, PREFIX_STATUS, PREFIX_RECRUITER_NAME, PREFIX_PRIORITY);
+                        PREFIX_DEADLINE, PREFIX_STATUS, PREFIX_RECRUITER_NAME, PREFIX_PRIORITY, PREFIX_REMARK);
```
Comment (tayruxin) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:42]: Wait so the user can add a remark from the add command?

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_REMARK + "[REMARK]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_REMARK + "Likes to swim.";
+    public static final String MESSAGE_REMOVE_USAGE = REMOVE_COMMAND_WORD
+            + ": Removes the remark identified by the index number used in the displayed company list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_NO_REMARK = "Remarks missing in entry.";
```
Comment (GlendaChong) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:35]: Maybe follow the current error messages style we have now?

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    public static final String MESSAGE_ADD_REMARK_SUCCESS = "Added remark to Company: %1$s";
+    public static final String MESSAGE_DELETE_REMARK_SUCCESS = "Removed remark from Company: %1$s";
+
+    private final Index index;
+    private final Remark remark;
+
+    /**
+     * @param index of the company in the filtered company list to edit the remark
+     * @param remark of the company to be updated to
+     */
```
Comment (GlendaChong) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:45]: Missing description of constructor. 

Should have fullstops after parameters description.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -186,4 +187,19 @@ public static Priority parsePriority(String priority) throws ParseException {
...
+     *
+     * @throws ParseException if the given {@code remark} is invalid.
+     */
+    public static Remark parseRemark(String remark) throws ParseException {
+        requireNonNull(remark);
+        String trimmedRemark = remark.trim();
+        if (!Remark.isValidRemark(trimmedRemark)) {
+            throw new ParseException(Remark.MESSAGE_CONSTRAINTS);
+        }
+        return new Remark(trimmedRemark);
```
Comment (GlendaChong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:203]: could add an additional exception for empty trimmedRemark

File: src/main/java/seedu/address/logic/parser/RemarkCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REMARK;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.RemarkCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.company.Remark;
+
+/**
+ * Parses input arguments and creates a new {@code RemarkCommand} object
```
Comment (GlendaChong) [inline src/main/java/seedu/address/logic/parser/RemarkCommandParser.java:14]: Missing full stop

File: src/main/java/seedu/address/logic/parser/RemarkCommandParser.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ */
+public class RemarkCommandParser implements Parser<RemarkCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the {@code RemarkCommand}
+     * and returns a {@code RemarkCommand} object for execution.
+     * @throws ParseException if the user input does not conform the expected format
```
Comment (GlendaChong) [inline src/main/java/seedu/address/logic/parser/RemarkCommandParser.java:20]: Missing full stop. Should have a space between description and @throw statement

File: src/main/java/seedu/address/logic/parser/UnremarkCommandParser.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.RemarkCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.company.Remark;
+
+
+/**
+ * Parses input arguments and creates a new {@code RemarkCommand} object
```
Comment (GlendaChong) [inline src/main/java/seedu/address/logic/parser/UnremarkCommandParser.java:13]: Missing full stop

**Review by GlendaChong — CHANGES_REQUESTED** [review]

**Pull request merged**